src/webdriver.nim

```diff
@@ -203,6 +213,106 @@ proc takeScreenshot*(self: Session): string =
+
+  return respObj["value"]
+
+proc execute*(self: Session, code: string, args: varargs[JsonNode]): JsonNode =
+  self.internalExecute(code, args, "sync")
+
+proc executeAsync*(self: Session, code: string, args: varargs[JsonNode]): JsonNode =
+  self.internalExecute(code, args, "async")
+
+proc addCookie*(self: Session, cookie: Cookie) =
+  let reqUrl = $(self.driver.url / "session" / self.id / "cookie")
+  let obj = %*  {
+    "cookie": {
+      "name": cookie.name,
+      "value": cookie.value,
+    }
+  }
+  if cookie.path.isSome:
+    obj["path"] = cookie.path.get.newJString()
+  if cookie.domain.isSome:
+    obj["domain"] = cookie.domain.get.newJString()
+  if cookie.secure.isSome:
+    obj["secure"] = cookie.secure.get.newJBool()
+  if cookie.httpOnly.isSome:
+    obj["httpOnly"] = cookie.httpOnly.get.newJBool()
+  if cookie.expiry.isSome:
+    obj["expiry"] = cookie.expiry.get.newJInt()
+
+  let resp = self.driver.client.post(reqUrl, $obj)
+  if resp.status != Http200:
+    raise newException(WebDriverException, resp.status)
+
+proc getCookie*(self: Session, name: string): Cookie =
+  let reqUrl = $(self.driver.url / "session" / self.id / "cookie" / name)
+
+  let resp = self.driver.client.get(reqUrl)
+
+  let cookie = checkResponse(resp.body)["value"]
+  result = Cookie(name: cookie["name"].getStr, value: cookie["value"].getStr)
+  if cookie.hasKey("path"):
+    result.path = some(cookie["path"].getStr)
+  if cookie.hasKey("domain"):
+    result.domain = some(cookie["domain"].getStr)
+  if cookie.hasKey("secure"):
+    result.secure = some(cookie["secure"].getBool)
+  if cookie.hasKey("httpOnly"):
+    result.httpOnly = some(cookie["httpOnly"].getBool)
+  if cookie.hasKey("expiry"):
+    result.expiry = some(cookie["expiry"].getBiggestInt)
+
+proc deleteCookie*(self: Session, name: string): Cookie =
+  let reqUrl = $(self.driver.url / "session" / self.id / "cookie" / name)
+
+  let resp = self.driver.client.delete(reqUrl)
+  if resp.status != Http200:
+    raise newException(WebDriverException, resp.status)
+
+proc getAllCookies*(self: Session): seq[Cookie] =
+  let reqUrl = $(self.driver.url / "session" / self.id / "cookie")
+
+  let resp = self.driver.client.get(reqUrl)
+  if resp.status != Http200:
+    raise newException(WebDriverException, resp.status)
+
+  let respObj = checkResponse(resp.body)
+  for cookie in respObj["value"].items:
+    var final = Cookie(name: cookie["name"].getStr, value: cookie["value"].getStr)
+    if cookie.hasKey("path"):
+      final.path = some(cookie["path"].getStr)
+    if cookie.hasKey("domain"):
+      final.domain = some(cookie["domain"].getStr)
+    if cookie.hasKey("secure"):
+      final.secure = some(cookie["secure"].getBool)
+    if cookie.hasKey("httpOnly"):
+      final.httpOnly = some(cookie["httpOnly"].getBool)
+    if cookie.hasKey("expiry"):
+      final.expiry = some(cookie["expiry"].getBiggestInt)
```

Comment:
There is some repetition here that could be refactored. Do so in a separate PR if you're up for it.